### PR TITLE
Basic profile page, basic update email/name

### PR DESCRIPTION
**Pull request by ksudea**
Basic profile page: /userProfile

Basic update form, with only email and name updates. No validation yet (in process).

### CODE REVIEW

File: app/models/userModel.py [hunk 1]
```diff
@@ -70,6 +75,54 @@ def register(email, password, firstname, lastname):
...
+        try:
+            rows = app.db.execute("""
+UPDATE Users
+SET firstname = :firstname
+WHERE id = :id
+""",
+                                  id=id, firstname = firstname)
+            return UserModel.get(id)
+        except Exception as e:
+            # likely email already in use; better error checking and reporting needed;
```
Comment (zsofiawalter) [inline app/models/userModel.py:105]: Error commenting here could be more specific

File: app/models/userModel.py [hunk 2]
```diff
@@ -70,6 +75,54 @@ def register(email, password, firstname, lastname):
...
+        try:
+            rows = app.db.execute("""
+UPDATE Users
+SET lastname = :lastname
+WHERE id = :id
+""",
+                                  id=id, lastname = lastname)
+            return UserModel.get(id)
+        except Exception as e:
+            # likely email already in use; better error checking and reporting needed;
```
Comment (zsofiawalter) [inline app/models/userModel.py:121]: Same as above

File: app/templates/base.html [hunk 3]
```diff
@@ -13,8 +13,9 @@
   <ul class="navbar">
     {% block login %}
       <img src="ThriftyCarrotMockLogo.png" alt="Cartoon Carrot Logo">
-      <h1 style="color:#ffffff">Thrifty Carrot</h1>
+      <h1 style="color:#ffffff" >Thrifty Carrot</h1>
       {% if current_user.is_authenticated %}
+        <a href="{{ url_for('users.profile') }}" type="button" class="btn btn-login">User Profile</a>
```
Comment (zsofiawalter) [inline app/templates/base.html:18]: Maybe we should leave the login/logout button and instead if the user is logged in a bunny icon/avatar shows up that the can press to go to their profile

File: app/users.py [hunk 4]
```diff
@@ -37,6 +37,41 @@ def login():
...
+    submit = SubmitField('Update')
+
+    def validate_email(self, email):
+        if UserModel.email_exists(email.data):
+            raise ValidationError('Already a user with this email.')
+
+    
+
+@bp.route('/userUpdate', methods=['GET', 'POST'])
+@login_required
```
Comment (zsofiawalter) [inline app/users.py:60]: This is good to know about! I didn't know that 'login_required' was a thing I could use haha

**Review by zsofiawalter — CHANGES_REQUESTED** [review]
Really great work! A few comments on being more specific in the exceptions in user model, and maybe even leave a few TODO tags if you have thoughts on next steps for exceptions
Also, note on user profile as an avatar button instead of a button that says user profile and leaving the log out button

File: app/models/userModel.py (hunk 1 above)
Comment (ksudea) [reply app/models/userModel.py:105]: Yep sounds good, I'll edit these along with the validations

**Review by zsofiawalter — APPROVED** [review]
Looks good to me, I like the changes to front end and this is a good start for our user page

**Pull request merged**